### ISSUE TITLE
Matured block in pool wallet, but not being transferred to miner's wallet

### ISSUE THREAD

**Issue by rbfx4x** [coinfoundry/miningcore#1676]
**Describe the bug**

Block is mined, and successfully transferred to pool wallet, but not to miner's wallet during payout window. Priv key and Pub key in config.json are double checked as correct. What does invalid signature mean?

[2023-04-22 01:36:55.4742] [I] [Bamboo] [Bamboo Payout Handler] Paying 50 BMB to 1 addresses 
[2023-04-22 01:36:55.4901] [I] [Bamboo] [Bamboo Payout Handler] [8638C87A42F27AD08B7EDEDBE73421A91423D0B4D8959A6A5B7CEEC5696EDEF0] Sending 50 BMB to 00837056D27EFCF2D3E2B3BAC5298DD623EEAACF2D9CC81513 
[2023-04-22 01:36:55.5259] [E] [Bamboo] [Bamboo Payout Handler] Failed to transfer the following balances: 00837056D27EFCF2D3E2B3BAC5298DD623EEAACF2D9CC81513 50 BMB: Payment transaction id: 8638C87A42F27AD08B7EDEDBE73421A91423D0B4D8959A6A5B7CEEC5696EDEF0 failed with error: INVALID_SIGNATURE

**To Reproduce**
1. Synced node and pool running normally
2. Miner finds a solo block
3. Block is credited to pool wallet after confirmation
4. Block reward will not transfer to miner (error above) and balance remains pending. 

**Expected behavior**
Block reward should be sent to miner. 

**Desktop (please complete the following information):**
 - OS: Ubuntu Server 20.0.4

**Additional context**
Relevant coin config for Pandanite (Bamboo)  
-----------------------------------------------------------
"bamboo": {
        "name": "Bamboo",
        "canonicalName": "Bamboo",
        "symbol": "BMB",
        "family": "bamboo",
        "website": "https://www.bamboocrypto.io/",
        "market": "",
        "twitter": "https://twitter.com/bmb_bamboo",
        "telegram": "",
        "discord": "https://discord.gg/2m9m2cnBk7",
        "coinbaseHasher": {
            "hash": "pufferfish2bmb"
        },
        "headerHasher": {
            "hash": "pufferfish2bmb"
        },
        "blockHasher": {
            "hash": "reverse",
            "args": [ { "hash": "pufferfish2bmb" } ]
        },
        "coinbaseIgnoreAuxFlags": true,
        "explorerBlockLink": "https://explorer.0xf10.com/block/$height$",
        "explorerTxLink": "https://explorer.0xf10.com/tx/{0}",
        "explorerAccountLink": "https://explorer.0xf10.com/account/{0}"
    },

config.json for "pools" with keys info redacted with XXX
------------------------------------------------------------------------
 "pools": [
        {
            "id": "Bamboo",
            "enabled": true,
            "coin": "bamboo",
            // Address to where block rewards are given (pool wallet)
            "address": "0003A9DC3803E12B7D71DC8BF853C9B900477B7D290512402E",
            "publicKey": "XXXXXX",
            "privateKey": "XXXXX",
                         "rewardRecipients": [
                {
                    "type": "op",
                    "address": "00475FD0E5FC59ACDB939DED13DD2B9E00E93479871585F5F1",
                    "percentage": 0.0
                }
            ],

**Comment by magecrimson** [comment]
I am pretty sure pufferfish is not supported in current build of miningcore.  There is a port which someone did which has the algo supported in limited aspects. @rbfx4x .

**Comment by rbfx4x** [comment]
That could be the issue. Is there a way I could jerryrig it a bit more to work as intended though?  
It's a personal solo pool at the moment so it's not a problem for miners to not get paid out since I also own the pool wallet :) But I'd like it to at least work fully.

**Comment by magecrimson** [comment]
No,  unless you want to add custom code and add a bunch of other things you must use the ported version of miningcore to get that coin to work.

I commented on your miningcore post as well.

Sent from my iPhone

On Apr 25, 2023, at 7:22 AM, rbfx4x ***@***.***> wrote:

﻿

That could be the issue. Is there a way I could jerryrig it a bit more to work as intended though?
It's a personal solo pool at the moment so it's not a problem for miners to not get paid out since I also own the pool wallet :) But I'd like it to at least work fully.
